Vizzuality/heco-invest
# Tag checkbox component

**In this PR** 

Added a `Forms/Tag` component
_Used as a regular uncontrolled checkbox_  
- Styling is achieved by means of using a `label`  
- Supports invalid states  
- Uses `useFocusRing` for keyboard navigation  

Added a `Forms/TagGroup` component  
_Responsible for standardising the visual styling of a group of tags_  
- Automatically displays a `Select All` button to select all tags  
- Can automatically set `react-hook-form`'s values and clear errors  

`project-developers/new` page  
  - Added the new `Tag` and `TagGroup` components  

&nbsp;

**Other relevant changes:**  
- Updated components that use category ids (eg: regular `Tag` and `TagsGrid` components) to use the correct ones the API will return, as well as a fifth category
  _Note: still hardcoded, not using enums. It was mostly to keep ids consistent on the frontend, and validate types_  
- Fixed an issue with the `MultiPageLayout` component which caused `useFocusRing`'s `isFocusVisible` to erroneously be set to `true` when clicking on tags. 

&nbsp;

**NOTE:** There's a small deviation from the acceptance criteria. Although Tags do take an `id` prop, it shouldn't be used to link to an external label. The reason for that is that was that to keep the tags behaving as an uncontrolled checkbox, and ensuring max compatibility with `react-hook-form` and `invalid` states, the label was embedded into the component. 

## Testing instructions

Respects the [visual design](https://www.figma.com/file/3epSYh4KQZBCyh4Y5OzmHw/HeCo---Design-System?node-id=2345%3A4757)

Has visual error state when input validation fails ([see error state example](https://www.figma.com/file/3epSYh4KQZBCyh4Y5OzmHw/HeCo---Design-System?node-id=2377%3A3138))

Compatible with React Hook Form (see [example of integration](https://react-hook-form.com/get-started/#Integratinganexistingform))

Accessible by [WCAG 2.1 standards](https://www.w3.org/TR/WCAG21/)

Focused state for keyboard users

Presented as a checkbox list to assisted technologies (AT)

Requires an 'id' prop to link to a label externally and a 'name' prop to link the checkboxes between themselves

Available in Storybook

## Tracking

[LET-170](https://vizzuality.atlassian.net/browse/LET-170)

## Screenshots
<img width="1040" alt="Screenshot 2022-04-12 at 12 56 09" src="https://user-images.githubusercontent.com/6273795/162957440-9760d2ff-06e6-46dc-9d4c-54bea75e4a60.png">
<img width="852" alt="Screenshot 2022-04-12 at 12 56 24" src="https://user-images.githubusercontent.com/6273795/162957448-ec831bed-8656-4615-8a95-cf487daee6b0.png">
